babel/dates.py

```diff
@@ -1282,6 +1284,16 @@ def format_timezone(self, char, num):
                 return get_timezone_name(self.value.tzinfo, width,
                                          uncommon=True, locale=self.locale)
             return get_timezone_location(self.value.tzinfo, locale=self.locale)
+        # Included additional elif condition to add support for 'X' in timezone format
+        elif char in "xX":
+            if get_timezone_gmt(self.value, width='short', locale=self.locale)=='+0000' and char == 'X':
```

Comment:
Hmm... It still feels kind of wasteful to have `get_timezone_gmt` go through actually formatting the date just so its return value could be turned into a zero :)

I think it might be better to actually add an argument to `get_timezone_gmt`, something like `return_z` (optional, false by default), which, when set to True, would cause its logic to short-circuit to returning `"Z"` directly after it has calculated that the offset would in fact be zero.

What do you think?